app/db/interface.py

```diff
@@ -71,8 +75,30 @@ def update_payment(self, payment_id, payment_replacement=None, payment_attribute
         :param payment_replacement: <dict> a complete payload that describes a new payload which replaces the old one
         :param payment_attributes: <dict> a collection of new payment attribute values that will overwrite old ones
         """
-
-        raise NotImplementedError()
+        payment = self.db.session.query(Payment).get(payment_id)
+        if payment == None or payment.is_removed == True :
+            raise InvalidPaymentID('Invalid payment: Payment ID not found',status_code=404)
+        if payment_replacement:
+            # TODO : test cases for validity in next sprint
+            #if not self.is_valid_put(payment.serialize(),payment_replacement):
+            #   raise DataValidationError('Invalid payment: body of request contained bad or no data')
+            payment.deserialize_put(payment_replacement)
+            self.db.session.commit()
+            return_object = payment.serialize()
+            return return_object
+        elif payment_attributes: #patch
+            existing_payment = payment.serialize()
+            for key in payment_attributes:
+                if key in UPDATABLE_PAYMENT_FIELDS:
+                    existing_payment[key] = payment_attributes[key]
+                else:
+                    raise DataValidationError('Invalid payment: body of request contains invalid/un-updatable fields')
+            payment.deserialize_put(existing_payment)
+            self.db.session.commit()
+            return_object = payment.serialize()
+            return return_object
+        else:
+            raise DataValidationError('Invalid payment: body of request contained bad or no data')
```

Comment:
some code repetition here that could be cleaned up. both `if` and `elif` blocks commit, serialize return object and return. we could do all of this once after `else`